src/mars.c

```diff
@@ -1531,6 +1531,20 @@ int main(int iargc, const char *argv[])
     {
         assert(global.params.moduleDepsFile != NULL);
 
+        // when generating depdencies we have to run sema3
+        // because otherwise we will miss some depdencies
+        // introdcued in function bodies
+        for (size_t i = 0; i < Module::amodules.dim; i++)
```

Comment:
Couldn't this start from 1 since the first module is the module being compiled and semantic has already been performed on it ?

Reply:
It could but who cares. I'd rather give no guarantees about order.
That's how bugs are born :)

Reply:
Please make it a `foreach` loop.

Reply:
Never mind.